### PR TITLE
DOCU-2926: DCR overview page and new authentication methods

### PR DESCRIPTION
**Pull request by acgoldsmith**
### Description

* adds new overview page for DCR including summary of available methods w/ new items bearer tokens and session cookies.
* new images to show options including bearer token UI
* reference links to IdP bearer token endpoints
* sourced from https://github.com/Kong/khcp/pull/5468 
* ref DOCU-2926

### Testing instructions

* needs SME review from @nateslo & team (Thank you). 
   Specifically, for the auth methods, are these commonly-used tasks that ppl know how to do and the endpoint references are sufficient?

* need to see how images render (needs prettier)

### Checklist 

- [ x] Review label added 
- [ x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

### CODE REVIEW

File: app/konnect/dev-portal/applications/dynamic-client-registration/overview.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
+title: DCR Overview
+content-type: explanation
+---
+
+## Dynamic Client Registration Overview
+
+Dynamic Client Registration (DCR)support in {{site.konnect_short_name}} provides multiple methods by which applications can be authenticated using industry-standard protocols.
```
Comment (Guaris) [inline app/konnect/dev-portal/applications/dynamic-client-registration/overview.md:8]:
```suggestion
Dynamic Client Registration (DCR) support in {{site.konnect_short_name}} provides multiple methods by which applications can be authenticated using industry-standard protocols.
```

**Review by sunnytoolsmiths — COMMENTED** [review]
Please update the side nav bar from portal auth to portal RBAC

In the API reference, please update portal auth API to portal rbac API

**Review by sunnytoolsmiths — CHANGES_REQUESTED** [review]
Please replace portal auth with portal rbac in the nav bar and API reference doc.

**Review by sunnytoolsmiths — CHANGES_REQUESTED** [review]
can we please also add the section about client credentials and session cookie

**Review by Mierenga — COMMENTED** [review]
@acgoldsmith Thanks for putting this info together, I've added a few suggestions that hopefully could clarify some things

**Review by acgoldsmith — COMMENTED** [review]
merging SME review, thanks @Mierenga

File: app/_data/docs_nav_konnect.yml [hunk 2]
```diff
@@ -151,6 +151,8 @@
           url: /dev-portal/applications/enable-app-reg
         - text: Dynamic Client Registration
           items:
+            - text: Overview
+              url: /dev-portal/applications/dynamic-client-registration/overview
```
Comment (lena-larionova) [inline app/_data/docs_nav_konnect.yml:155]: Can you please rename the file from `overview.md` to `index.md`? Using "overview" lengthens the URL unnecessarily.

File: app/konnect/dev-portal/applications/dynamic-client-registration/overview.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+title: Dynamic Client Registration Overview
+content-type: explanation
+---
+
+Dynamic Client Registration (DCR) within {{site.konnect_short_name}} Dev Portal allows applications created in the portal to automatically create a linked application in a third-party Identity Provider (IdP).
+This outsources the issuer and management of application credentials to a third party, allowing for additional configuration options and compatibility with various OIDC features provided by the IdP.
+
+## Authentication Methods
+
+DCR support in {{site.konnect_short_name}} provides multiple methods by which applications can be authenticated using industry-standard protocols. These methods include:
```
Comment (cloudjumpercat) [inline app/konnect/dev-portal/applications/dynamic-client-registration/overview.md:11]:
```suggestion
{{site.konnect_short_name}} provides multiple methods by which applications can be authenticated using industry-standard protocols. These methods include:
```
Just a suggestion, I think either one gets the meaning across.

File: app/konnect/dev-portal/applications/dynamic-client-registration/overview.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+* **Client Credentials Grant**: Authenticate with the client ID and secret provided to the application.
+* **Bearer tokens**: Authenticate using a token requested from the IdP's `/token` endpoint.
+* **Session cookie**: Allow sessions from either client credentials or bearer tokens to persist via cookie until an expiration.
```
Comment (cloudjumpercat) [inline app/konnect/dev-portal/applications/dynamic-client-registration/overview.md:14]: Is the expiration configured by the customer? Or is it a default set by the cookie/Konnect?

File: app/_data/docs_nav_konnect.yml (hunk 2 above)
Comment (lena-larionova) [inline app/_data/docs_nav_konnect.yml:155]:
```suggestion
              url: /dev-portal/applications/dynamic-client-registration/
```
once the file is renamed, can shorten this URL.

**Review by cloudjumpercat — APPROVED** [review]
Left a few nitpick suggestions, but everything else looks really good!

**Pull request merged**